lib/logsource.c

```diff
@@ -259,10 +270,10 @@ log_source_post(LogSource *self, LogMessage *msg)
   log_msg_add_ack(msg, &path_options);
   msg->ack_func = log_source_msg_ack;
 
-  old_window_size = g_atomic_counter_exchange_and_add(&self->window_size, -1);
+  old_window_size = window_size_counter_sub(&self->window_size, 1, NULL);
 
   if (G_UNLIKELY(old_window_size == 1))
-    msg_debug("Source has been suspended", log_pipe_location_tag(&self->super));
+    msg_debug("Source has been suspended", log_pipe_location_tag(&self->super), evt_tag_str("Function", __FUNCTION__));
```

Comment:
Please consider adding this new evt tag to `log_source_flow_control_suspend` as well. We have the same debug message there.

I don't know about our evt tag conventions, but it seems we prefer lowercase tag names (`function` vs. `Function`).

Reply:
ok, will check it.